src/eve-server/system/DestinyManager.cpp

```diff
@@ -1648,25 +1737,49 @@ void DestinyManager::WarpUpdate(double currentShipSpeed) {
     m_velocity = (m_warpState->warp_vector * currentShipSpeed);
     SetPosition(m_targetPoint - (m_warpState->warp_vector * m_targetDistance));
 
-    if (m_warpState->decel) {
-        if (mySE->SysBubble() == nullptr) {
-            if (is_log_enabled(DESTINY__WARP_TRACE))
-                _log(DESTINY__WARP_TRACE, "Destiny::WarpUpdate()  %s(%u): Ship is %f from center of target bubble %u.",\
-                        mySE->GetName(), mySE->GetID(), m_targBubble->GetCenter().distance(m_position), m_targBubble->GetID());
-            if (m_targBubble->InBubble(m_position, true)) {
-                if (is_log_enabled(DESTINY__WARP_TRACE))
-                    _log(DESTINY__WARP_TRACE, "Destiny::WarpUpdate()  %s(%u): Ship at %.2f,%.2f,%.2f is calling Add() for bubble %u.", \
-                            mySE->GetName(), mySE->GetID(), m_position.x, m_position.y, m_position.z, m_targBubble->GetID());
-                m_targBubble->Add(mySE);
-                SetPosition(m_position, true);
-            }
+    if (is_log_enabled(DESTINY__WARP_TRACE)) {
+        _log(
+            DESTINY__WARP_TRACE,
+            "Destiny::WarpUpdate()  %s(%u): Ship is %f from center of target bubble %u.",
+            mySE->GetName(),
+            mySE->GetID(),
+            m_targBubble->GetCenter().distance(m_position),
+            m_targBubble->GetID()
+        );
+    }
+
+    if (m_targBubble->InBubble(m_position, true)) {
+        if (is_log_enabled(DESTINY__WARP_TRACE)) {
+            _log(
+                DESTINY__WARP_TRACE,
+                "Destiny::WarpUpdate()  %s(%u): Ship at %.2f,%.2f,%.2f is calling Add() for bubble %u.",
+                mySE->GetName(),
+                mySE->GetID(),
+                m_position.x,
+                m_position.y,
+                m_position.z,
+                m_targBubble->GetID()
+            );
         }
+
+        m_targBubble->Add(mySE);
+
+        SetPosition(m_position, true);
+    } else {
+        _log(
+            DESTINY__WARP_TRACE,
+            "Destiny::WarpUpdate()  %s(%u): adding to midWarpSystemBubble.",
+            mySE->GetName(),
+            mySE->GetID()
+        );
+        SystemBubble* midWarpSystemBubble(sBubbleMgr.GetBubble(mySE->SystemMgr(), m_position));
+        midWarpSystemBubble->Add(mySE);
```

Comment:
Here is an actual code change. A sysbubble is created each server tick during warp, which fixes a few issues that arise because the ship is in space but before my code changes, it gets removed from the bubble it was in when starting warp, and its sysbubble attribute is a `nullptr`.

I have also tested this in a 2-person fleet warp. The other person _should_ be in the same bubble (unable to verify due to noisy log output and no player ids in logs), but unfortunately, at this time the client does not show the other person once warp cruise has been reached. Maybe this can be addressed in the future.